go-update/src/main/java/org/reactome/release/goupdate/DuplicateCleaner.java

```diff
@@ -0,0 +1,215 @@
+				for (String reactomeClass : Arrays.asList(ReactomeJavaConstants.GO_BiologicalProcess, ReactomeJavaConstants.GO_MolecularFunction, ReactomeJavaConstants.GO_CellularComponent))
+				{
+					@SuppressWarnings("unchecked")
+					Collection<GKInstance> instances = (Collection<GKInstance>) adaptor.fetchInstanceByAttribute(reactomeClass, ReactomeJavaConstants.accession, "=", accession);
+					// For each instance for this accession, try to find the newest one so that it won't be deleted.
+					if (instances != null && instances.size() > 0)
+					{
+						// Add all the DB_IDs to the list to delete. Later, only the DB_ID of the NEWEST instance will be removed from this list, and all other older instances will be deleted.
+						dbIDsToDelete.addAll(instances.stream().map(i -> i.getDBID()).collect(Collectors.toSet()));
+						newestInstance = findNewestInstance(instances);
+					}
+				}
+				logger.debug("Newest instance for accession {} is {} - this instance will NOT be deleted.", accession, newestInstance.toString());
+				// REMOVE the ID of the newest instance from the list of things to delete (it would have been automatically added if the "newest" instance
+				// had 0 referrers).
+				dbIDsToDelete.remove(newestInstance.getDBID());
+			}
+		}
+		logger.info("\n\nSummary:\nTotal number of duplicated accessions: {} \n"
+				+ "Number of instances with significant (non-GO Term) referrers: {}\n", countOfDuplicatesAccessions, instancesWithSignificantReferrers);
+		logger.info("{} IDs will be deleted.", dbIDsToDelete.size());
+		adaptor.startTransaction();
+		for (Long dbID : dbIDsToDelete)
+		{
+			GKInstance instance = adaptor.fetchInstance(dbID);
+			logger.info("DB ID {} (for accession {}) will be deleted", dbID, instance.getAttributeValue(ReactomeJavaConstants.accession));
+			if (!this.testMode)
+			{
+				adaptor.deleteByDBID(dbID);
+			}
+		}
+		adaptor.commit();
+	}
+
+	/**
+	 * Finds the newest GKInstance in a list of instances.
+	 * @param instances - The list of instances
+	 * @return The newest instance.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private GKInstance findNewestInstance(Collection<GKInstance> instances) throws InvalidAttributeException, Exception
+	{
+		GKInstance newestInstance = null;
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.S");
+		LocalDateTime newestDate = null;
+		for (GKInstance instance : instances)
+		{
+			LocalDateTime localDate = null;
+			GKInstance modifiedInstance = null;
+			modifiedInstance = (GKInstance) instance.getAttributeValue(ReactomeJavaConstants.modified);
+			
+			if (modifiedInstance != null)
+			{
+				localDate = LocalDateTime.parse((CharSequence) modifiedInstance.getAttributeValue(ReactomeJavaConstants.dateTime), formatter);
+			}
+			else
+			{
+				modifiedInstance = (GKInstance) instance.getAttributeValue(ReactomeJavaConstants.created);
```

Comment:
I think naming the returned 'created' attribute 'modifiedInstance' is confusing, since there is the 'modified' attribute

Reply:
Nevermind, I see that you are just repurposing the above 'modifiedInstance' variable